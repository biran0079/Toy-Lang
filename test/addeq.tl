a=1;
b=a;
a+=1;
print(a);
print(b);
a=[];
b=a;
a+=[2];
print(a);
print(b);
a=a+[3];
print(a);
print(b);
a[0]+=1;
print(a);
c=[a];
c+=[b];
print(c);
c[1]+=[4];
print(c);
print(b);
c+=[1,2];
print(c);
