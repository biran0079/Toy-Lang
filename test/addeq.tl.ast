[stmts, [=, [id, a], [int, 1]], [=, [id, b], [id, a]], [+=, [id, a], [int, 1]], [call, [id, print], [exps, [id, a]]], [call, [id, print], [exps, [id, b]]], [=, [id, a], [list]], [=, [id, b], [id, a]], [+=, [id, a], [list, [int, 2]]], [call, [id, print], [exps, [id, a]]], [call, [id, print], [exps, [id, b]]], [=, [id, a], [+, [id, a], [list, [int, 3]]]], [call, [id, print], [exps, [id, a]]], [call, [id, print], [exps, [id, b]]], [+=, [list_access, [id, a], [int, 0]], [int, 1]], [call, [id, print], [exps, [id, a]]], [=, [id, c], [list, [id, a]]], [+=, [id, c], [list, [id, b]]], [call, [id, print], [exps, [id, c]]], [+=, [list_access, [id, c], [int, 1]], [list, [int, 4]]], [call, [id, print], [exps, [id, c]]], [call, [id, print], [exps, [id, b]]], [+=, [id, c], [list, [int, 1], [int, 2]]], [call, [id, print], [exps, [id, c]]]]
