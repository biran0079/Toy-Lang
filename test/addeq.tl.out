2
1
[2]
[2]
[2, 3]
[2]
[3, 3]
[[3, 3], [2]]
[[3, 3], [2, 4]]
[2, 4]
[[3, 3], [2, 4], 1, 2]
